### PR TITLE
[fix](union) union source operator hold too many blocks from children

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

File: be/src/pipeline/exec/data_queue.cpp [hunk 1]
```diff
@@ -146,6 +151,10 @@ void DataQueue::push_block(std::unique_ptr<vectorized::Block> block, int child_i
         if (_source_dependency) {
             set_source_ready();
         }
+
+        if (_spill_dependencies[child_idx]) {
```
Comment (yiguolei) [inline be/src/pipeline/exec/data_queue.cpp:155]: 我们不应当修改data queue的结构，dataqueue 是一个核心的数据结构，他只应该负责queue的逻辑。设想一下，假如我们的data queue 是使用的第三方的实现怎么办？

**Comment by mrhhsg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Comment by mrhhsg** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**